models/article-models.js

```diff
@@ -1,13 +1,102 @@
 const db = require('../db/connection'); 
 
+const checkTopicExists = (term) => {
+    return db.query(`SELECT * FROM topics WHERE slug = $1`, [term])
+    .then(({ rows: [topic] }) => {
+        if (!topic) {
+            return Promise.reject({
+                status: 404, 
+                msg: "Topic not found"
+            })
+        }
+    })
+}
+
+const fetchArticles = (sortOn = 'created_at', order = 'DESC', term) => {
+    let basicQuery = `SELECT articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_id, 
+    (SELECT COUNT(comment_id) :: INT) AS comment_count 
+     FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id `
+     let basicQuery2 = `
+     GROUP BY articles.article_id`;
+
+     if (!["title", "author", "article_id", "created_at", "votes", "comment_count", "topic"].includes(sortOn)) {
+        return Promise.reject({
+            status: 400, 
+            msg: "Invalid sort query"
+        }); 
+    }
+    if (!["ASC", "DESC"].includes(order)) {
+        return Promise.reject({
+            status: 400,
+            msg: "Invalid order query"
+        })
+    }
+    
+    switch (sortOn) {
```

Comment:
Not a big fan of using switch here only because it is soooo long and doesn't really make your code dry. If you have time later on, you should refactor this and maybe use template literals instead. But as long as it's working now you should be ok.